### ISSUE TITLE
Automatic updates are not working

### ISSUE THREAD

**Issue by hamamotsu** [aiplugs/poshapp#7]
![20200630](https://user-images.githubusercontent.com/4524950/86106631-c0a7c700-bafb-11ea-94fc-b516c7b85ab0.gif)

Auto-update does not start when I start up.
Posh APP version is 0.9.20.

**Comment by hamamotsu** [comment]
The startup log.

```
PS C:\Users\hoge-user\AppData\Local\Programs\electron.net.host> Version 0.9.20
 Electron Socket IO Port: 8000
Electron Socket started on port 8000 at 127.0.0.1
ASP.NET Core Port: 8001
stdout: Use Electron Port: 8000

stdout: info: Microsoft.Hosting.Lifetime[0]
      Now listening on: http://127.0.0.1:8001

stdout: info: Microsoft.Hosting.Lifetime[0]
      Application started. Press Ctrl+C to shut down.
info: Microsoft.Hosting.Lifetime[0]
      Hosting environment: Production
info: Microsoft.Hosting.Lifetime[0]
      Content root path: C:\Users\hoge-user\AppData\Local\Programs\electron.net.host\resources\bin\

ASP.NET Core Application connected... global.electronsocket uisIi700YLbQmqkDAAAA 2020-06-30T09:12:20.236Z
stdout: BridgeConnector connected!

Checking for update
Found version 0.9.43 (url: POSH-App-Setup-0.9.43.exe)
Downloading update from POSH-App-Setup-0.9.43.exe
updater cache dir: C:\Users\hoge-user\AppData\Local\electron.net.host-updater
No cached update info available
Download block maps (old: "https://github.com/aiplugs/poshapp/releases/download/v0.9.20/POSH-App-Setup-0.9.20.exe.blockmap", new: https://github.com/aiplugs/poshapp/releases/download/v0.9.43/POSH-App-Setup-0.9.43.exe.blockmap)
064WUUrBmIYTO+Jbi2tw9Q4X duplicated in blockmap (same size), it doesn't lead to broken differential downloader, just corresponding block will be skipped)
YnGKzm2jRqT6fmdkroQm4/xM duplicated in blockmap (same size), it doesn't lead to broken differential downloader, just corresponding block will be skipped)
fr9d4+kQiHy9J/HOF0FSSbXo duplicated in blockmap (same size), it doesn't lead to broken differential downloader, just corresponding block will be skipped)
q3x7VAMepRb1FFOAVMEy48uR duplicated in blockmap (same size), it doesn't lead to broken differential downloader, just corresponding block will be skipped)
ALZOZlIet3LJ0EGD7612BuFC duplicated in blockmap (same size), it doesn't lead to broken differential downloader, just corresponding block will be skipped)
M/IQigHV6/XTscL2A/zyAIEG duplicated in blockmap (same size), it doesn't lead to broken differential downloader, just corresponding block will be skipped)
NSDKH/YnSoKq7hVBW90TiO4m duplicated in blockmap (same size), it doesn't lead to broken differential downloader, just corresponding block will be skipped)
IW4u55tiA1sK0NOsw7t6bNmA duplicated in blockmap (same size), it doesn't lead to broken differential downloader, just corresponding block will be skipped)
esDEWq4ApTBIfIk5x6V66Oyh duplicated in blockmap (same size), it doesn't lead to broken differential downloader, just corresponding block will be skipped)
GzNaJ5mALtl9yRj3YFxWcwez duplicated in blockmap (same size), it doesn't lead to broken differential downloader, just corresponding block will be skipped)
iMwEK/iUXgnGBu07jXiyXWzv duplicated in blockmap (same size), it doesn't lead to broken differential downloader, just corresponding block will be skipped)
25d+xByrZX9f2tOFMfAXmwCz duplicated in blockmap (same size), it doesn't lead to broken differential downloader, just corresponding block will be skipped)
cR4hZjHDDx1zHEmwuDXu7lYQ duplicated in blockmap (same size), it doesn't lead to broken differential downloader, just corresponding block will be skipped)
SSqPWn5FUUkFbpN2Z7yZBoMY duplicated in blockmap (same size), it doesn't lead to broken differential downloader, just corresponding block will be skipped)
W58XJdJGSlxiNvt02yUJctqm duplicated in blockmap (same size), it doesn't lead to broken differential downloader, just corresponding block will be skipped)
2+vu1PiIp3RbsAfWGIGbNw2P duplicated in blockmap (same size), it doesn't lead to broken differential downloader, just corresponding block will be skipped)
File has 5262 changed blocks
[
  {
    "kind": 1,
    "start": 0,
    "end": 79097
  },
  {
    "kind": 0,
    "start": 87304,
    "end": 133790
  },
  {
    "kind": 1,
    "start": 125583,
    "end": 6946249
  },
  {
    "kind": 0,
    "start": 6915600,
    "end": 6929455
  },
  {
    "kind": 1,
    "start": 6960104,
    "end": 6975657
  },
  {
    "kind": 0,
    "start": 6945011,
    "end": 6955808
  },
  {
    "kind": 1,
    "start": 6986454,
    "end": 7691628
  },
  {
    "kind": 0,
    "start": 7630583,
    "end": 7659532
  },
  {
    "kind": 1,
    "start": 7720577,
    "end": 9922678
  },
  {
    "kind": 0,
    "start": 9761098,
    "end": 10173325
  },
  {
    "kind": 1,
    "start": 10334905,
    "end": 10367673
  },
  {
    "kind": 0,
    "start": 10206093,
    "end": 10231258
  },
  {
    "kind": 1,
    "start": 10392838,
    "end": 11066815
  },
  {
    "kind": 0,
    "start": 10905926,
    "end": 11119703
  },
  {
    "kind": 1,
    "start": 11280592,
    "end": 11311175
  },
  {
    "kind": 0,
    "start": 11150287,
    "end": 11246804
  },
  {
    "kind": 1,
    "start": 11407692,
    "end": 11548308
  },
  {
    "kind": 0,
    "start": 12213316,
    "end": 12268123
  },
  {
    "kind": 1,
    "start": 11603115,
    "end": 11631931
  },
  {
    "kind": 0,
    "start": 12289202,
    "end": 12306713
  },
  {
    "kind": 1,
    "start": 11649442,
    "end": 11669328
  },
  {
    "kind": 0,
    "start": 12326595,
    "end": 12335609
  },
  {
    "kind": 1,
    "start": 11678342,
    "end": 11700807
  },
  {
    "kind": 0,
    "start": 12358071,
    "end": 13591249
  },
  {
    "kind": 1,
    "start": 12933985,
    "end": 12963114
  },
  {
    "kind": 0,
    "start": 13620348,
    "end": 15743510
  },
  {
    "kind": 1,
    "start": 15086276,
    "end": 15262685
  },
  {
    "kind": 0,
    "start": 15950348,
    "end": 17327520
  },
  {
    "kind": 1,
    "start": 16639857,
    "end": 50214793
  },
  {
    "kind": 0,
    "start": 50395280,
    "end": 50807802
  },
  {
    "kind": 1,
    "start": 50627315,
    "end": 54391976
  },
  {
    "kind": 0,
    "start": 54279568,
    "end": 54735083
  },
  {
    "kind": 1,
    "start": 54847491,
    "end": 67446322
  },
  {
    "kind": 0,
    "start": 63768396,
    "end": 64231030
  },
  {
    "kind": 1,
    "start": 67908956,
    "end": 71855094
  },
  {
    "kind": 0,
    "start": 67430380,
    "end": 68149930
  },
  {
    "kind": 1,
    "start": 72574644,
    "end": 81070338
  },
  {
    "kind": 0,
    "start": 75472575,
    "end": 75516932
  },
  {
    "kind": 1,
    "start": 81114695,
    "end": 111768221
  },
  {
    "kind": 0,
    "start": 101908938,
    "end": 102220949
  },
  {
    "kind": 1,
    "start": 112080232,
    "end": 117587248
  },
  {
    "kind": 0,
    "start": 103470635,
    "end": 103702067
  },
  {
    "kind": 1,
    "start": 117818680,
    "end": 117956504
  }
]
Full: 115,191.9 KB, To download: 107,085.81 KB (93%)
Differential download: https://github.com/aiplugs/poshapp/releases/download/v0.9.43/POSH-App-Setup-0.9.43.exe
download range: bytes=0-79096
Redirect to https://github-production-release-asset-2e65be.s3.amazonaws.com/229247001/88c55480-ba13-11ea-9764-815d51787638
download range: bytes=125583-6946248
Install on explicit quitAndInstall
Error: Error: No valid update available, can't quit and install
    at NsisUpdater.install (C:\Users\hoge-user\AppData\Local\Programs\electron.net.host\resources\app.asar\node_modules\electron-updater\out\BaseUpdater.js:61:26)
    at NsisUpdater.quitAndInstall (C:\Users\hoge-user\AppData\Local\Programs\electron.net.host\resources\app.asar\node_modules\electron-updater\out\BaseUpdater.js:28:30)
    at C:\Users\hoge-user\AppData\Local\Programs\electron.net.host\resources\app.asar\api\autoUpdater.js:101:40
    at Generator.next (<anonymous>)
    at C:\Users\hoge-user\AppData\Local\Programs\electron.net.host\resources\app.asar\api\autoUpdater.js:8:71
    at new Promise (<anonymous>)
    at __awaiter (C:\Users\hoge-user\AppData\Local\Programs\electron.net.host\resources\app.asar\api\autoUpdater.js:4:12)
    at Socket.<anonymous> (C:\Users\hoge-user\AppData\Local\Programs\electron.net.host\resources\app.asar\api\autoUpdater.js:100:75)
    at Socket.emit (events.js:203:13)
    at C:\Users\hoge-user\AppData\Local\Programs\electron.net.host\resources\app.asar\node_modules\socket.io\lib\socket.js:528:12
download range: bytes=6960104-6975656
download range: bytes=6986454-7691627
download range: bytes=7720577-9922677
download range: bytes=10334905-10367672
download range: bytes=10392838-11066814
download range: bytes=11280592-11311174
download range: bytes=11407692-11548307
download range: bytes=11603115-11631930
download range: bytes=11649442-11669327
download range: bytes=11678342-11700806
download range: bytes=12933985-12963113
download range: bytes=15086276-15262684
download range: bytes=16639857-50214792
download range: bytes=50627315-54391975
download range: bytes=54847491-67446321
download range: bytes=67908956-71855093
download range: bytes=72574644-81070337
download range: bytes=81114695-111768220
Cannot download differentially, fallback to full download: Error: Request timed out
    at Timeout.timeoutCb [as _onTimeout] (C:\Users\hoge-user\AppData\Local\Programs\electron.net.host\resources\app.asar\node_modules\builder-util-runtime\out\httpExecutor.js:319:16)
    at listOnTimeout (internal/timers.js:531:17)
    at processTimers (internal/timers.js:475:7)
ASP.NET Core Application connected... global.electronsocket bwvGKS9Hg2VRwCfQAAAB 2020-06-30T09:13:21.577Z
stdout: BridgeConnector connected!

Error: Error: Request timed out
    at Timeout.timeoutCb [as _onTimeout] (C:\Users\hoge-user\AppData\Local\Programs\electron.net.host\resources\app.asar\node_modules\builder-util-runtime\out\httpExecutor.js:319:16)
    at listOnTimeout (internal/timers.js:531:17)
    at processTimers (internal/timers.js:475:7)
(node:13380) UnhandledPromiseRejectionWarning: Error: Request timed out
    at Timeout.timeoutCb [as _onTimeout] (C:\Users\hoge-user\AppData\Local\Programs\electron.net.host\resources\app.asar\node_modules\builder-util-runtime\out\httpExecutor.js:319:16)
    at listOnTimeout (internal/timers.js:531:17)
    at processTimers (internal/timers.js:475:7)
(node:13380) UnhandledPromiseRejectionWarning: Error: Request timed out
    at Timeout.timeoutCb [as _onTimeout] (C:\Users\hoge-user\AppData\Local\Programs\electron.net.host\resources\app.asar\node_modules\builder-util-runtime\out\httpExecutor.js:319:16)
    at listOnTimeout (internal/timers.js:531:17)
    at processTimers (internal/timers.js:475:7)
(node:13380) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:13380) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:13380) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
(node:13380) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

```

**Comment by iwate** [comment]
@hamamotsu Thx!

**Comment by iwate** [comment]
fix it at #10
